frappe/desk/page/leaderboard/leaderboard.js

```diff
@@ -0,0 +1,362 @@
+				return {"label": __(frappe.model.unscrub(d)), value: d };
+			})
+		);
+
+		this.timespan_select.on("change", (e) => {
+			this.options.selected_timespan = e.currentTarget.value;
+			this.make_request();
+		});
+
+		this.type_select.on("change", (e) => {
+			this.options.selected_filter_item = e.currentTarget.value;
+			this.make_request();
+		});
+	}
+
+	render_selected_doctype() {
+
+		this.$sidebar_list.on("click", "li", (e)=> {
+			let $li = $(e.currentTarget);
+			let doctype = $li.find("span").attr("doctype-value");
+
+			this.options.selected_company = frappe.defaults.get_default("company");
+			this.options.selected_doctype = doctype;
+			this.options.selected_filter = this.filters[doctype];
+			this.options.selected_filter_item = this.filters[doctype][0];
+
+			this.type_select.empty().add_options(
+				this.options.selected_filter.map(d => {
+					return {"label": __(frappe.model.unscrub(d)), value: d };
+				})
+			);
+			if (this.leaderboard_config[this.options.selected_doctype].company_disabled) {
+				$(this.parent).find("[data-original-title=Company]").hide();
+			} else {
+				$(this.parent).find("[data-original-title=Company]").show();
+			}
+
+			this.$sidebar_list.find("li").removeClass("active");
+			$li.addClass("active");
+
+			frappe.set_route("leaderboard", this.options.selected_doctype);
+			this.make_request();
+		});
+	}
+
+	render_search_box() {
+
+		this.$search_box =
+			$(`<div class="leaderboard-search col-md-3">
+				<input type="text" placeholder="Search" class="form-control leaderboard-search-input input-sm">
+			</div>`);
+
+		$(this.parent).find(".page-form").append(this.$search_box);
+	}
+
+	setup_search(list_items) {
+		let $search_input = this.$search_box.find(".leaderboard-search-input");
+
+		this.$search_box.on("keyup", ()=> {
+			let text_filter = $search_input.val().toLowerCase();
+			text_filter = text_filter.replace(/^\s+|\s+$/g, '');
+			for (var i = 0; i < list_items.length; i++) {
+				let text = list_items.eq(i).find(".list-id").text().trim().toLowerCase();
+
+				if (text.includes(text_filter)) {
+					list_items.eq(i).css("display", "");
+				} else {
+					list_items.eq(i).css("display", "none");
+				}
+			}
+		});
+	}
+
+	show_leaderboard(doctype) {
+		if (this.doctypes.length) {
+			if (this.doctypes.includes(doctype)) {
+				this.options.selected_doctype = doctype;
+				this.$sidebar_list.find(`[doctype-value = "${this.options.selected_doctype}"]`).trigger("click");
+			}
+
+			this.$search_box.find(".leaderboard-search-input").val("");
+			frappe.set_route("leaderboard", this.options.selected_doctype);
+		}
+	}
+
+	make_request() {
+
+		frappe.model.with_doctype(this.options.selected_doctype, ()=> {
+			this.get_leaderboard(this.get_leaderboard_data);
+		});
+	}
+
+	get_leaderboard(notify) {
+		if (!this.options.selected_company) {
+			frappe.throw(__("Please select Company"));
+		}
+
+		frappe.call({
+			method: "frappe.desk.page.leaderboard.leaderboard.get_leaderboards",
```

Comment:
Can't we directly call the method in the leaderboard config instead of sending complete leaderboard config back to the server?

`frappe.xcall(this.leaderboard_config.method, args)`